src/shogun/lib/parameter_observers/ObservedValue.cpp

```diff
@@ -0,0 +1,15 @@
+/*
+* Written (W) 2019 Giovanni De Toni
+*/
+
+#include <shogun/base/Parameter.h>
+#include <shogun/lib/parameter_observers/ObservedValue.h>
+
+using namespace shogun;
+
+ObservedValue::ObservedValue(int64_t step, char * name)
+	: CSGObject(), m_step(step), m_name(name)
+{
+	SG_ADD(&step, "step", "Step", ParameterProperties::NONE);
+	SG_ADD(name, "name", "Name of the observed value", ParameterProperties::NONE);
```

Comment:
Here `name` is actually a `char *`. From what I could see, `std::string` is not accepted by the parameter framework (?).

Reply:
Check the definition of the macro
But I think we can add something to use string here as well

Reply:
The class `Parameter` does not have any method to store an `std::string` object. I guess we could add it. Should it be straightforward, right?

Reply:
Let’s not touch the old parameter code. Instead, don’t use sgadd here but rather register with the watch method used inside sgadd. This means that the parameter will not be serialised for now but that’s fine for these observer classes imo.
You *can* register strings with the new framework